frontend/sw360-portlet/src/main/java/com/siemens/sw360/portal/portlets/licenses/LicensesPortlet.java

```diff
@@ -155,6 +197,81 @@ private void addLicenseBreadcrumb(RenderRequest request, RenderResponse response
+
+        if (!Strings.isNullOrEmpty(licenseId)) {
+            try {
+                license = client.getByID(licenseId, user.getDepartment());
+            } catch (TException e) {
+                log.error("Error updating license:", e);
+            }
+        }
+
+        license = updateLicenseFromRequest(license,request);
+
+        RequestStatus requestStatus=RequestStatus.FAILURE;
+        try {
+            requestStatus = client.updateLicense(license, user, user);
+        } catch (TException e) {
+            log.error("Could not add or update license:" + e);
+        }
+        if (licenseId != null) {
+            response.setRenderParameter(LICENSE_ID, licenseId);
+            response.setRenderParameter(PAGENAME, PAGENAME_DETAIL);
+            response.setRenderParameter(SELECTED_TAB, "Details");
+            setSessionMessage(request, requestStatus, "License", "update");
+        } else {
+            response.setRenderParameter(PAGENAME, PAGENAME_VIEW);
+            setSessionMessage(request, requestStatus, "License", "adde");
+        }
+    }
+
+    private License updateLicenseFromRequest(License license, ActionRequest request) {
+        String text = request.getParameter(License._Fields.TEXT.name());
+        String fullname = request.getParameter(License._Fields.FULLNAME.name());
+        String shortname = request.getParameter(License._Fields.SHORTNAME.name());
+        boolean gpl2compatibility =
+                (request.getParameter(License._Fields.GPLV2_COMPAT.toString()) == null) ? false : true;
+        boolean gpl3compatibility =
+                (request.getParameter(License._Fields.GPLV3_COMPAT.toString()) == null) ? false : true;
+        String licenseTypeString =
+                request.getParameter(License._Fields.LICENSE_TYPE.toString() + LicenseType._Fields.LICENSE_TYPE.toString());
+        license.setText(CommonUtils.nullToEmptyString(text));
+        license.setFullname(CommonUtils.nullToEmptyString(fullname));
+        license.setShortname((CommonUtils.nullToEmptyString(shortname)));
+        license.setGPLv2Compat(gpl2compatibility);
+        license.setGPLv3Compat(gpl3compatibility);
+        try {
+            String licenseTypeDatabaseId = getDatabaseIdFromLicenseType(licenseTypeString);
+            license.setLicenseTypeDatabaseId(licenseTypeDatabaseId);
+        } catch (TException e) {
+            log.error("Could not set licenseTypeDatabaseId:" + e);
+        }
+        return license;
+    }
+
+    private String getDatabaseIdFromLicenseType(String licenseTypeString) throws TException {
+        if (licenseTypes == null) {
+            LicenseService.Iface client = thriftClients.makeLicenseClient();
+            try {
+                licenseTypes = client.getLicenseTypes();
+            } catch (TException e){
+                throw new SW360Exception("Error getting license type list:"+ e);
+            }
+        }
+        for (LicenseType licenseType : licenseTypes) {
```

Comment:
I prefer an Exception, since the licenseTypeString really should be in the list - the user can only choose between options from the list. Given the exception approach, the given code is shorter.